src/insert.js

```diff
@@ -23,5 +21,7 @@ var _slice = require('./internal/_slice');
  */
 module.exports = _curry3(function insert(idx, elt, list) {
   idx = idx < list.length && idx >= 0 ? idx : list.length;
-  return _concat(_append(elt, _slice(list, 0, idx)), _slice(list, idx));
+  var result = _slice(list);
+  result.splice(idx, 0, elt);
+  return result;
```

Comment:
I thought we tested that approach and the `concat` one was faster.  But it's only vague memories now, I'm afraid.